output/output.go

```diff
@@ -41,11 +41,23 @@ const (
 	JsonIndent = "    "
 )
 
-type printRequest struct {
-	cmd         *cobra.Command
-	format      string
-	fields      string
-	annotations map[string]string
+type Table struct {
+	// table output
+	Headers []string
+	Lines   [][]string
+	Detail  bool // true to print a single line as a name: value multi-line instead of table
+
+	// extract field columns in the same order as headers
+	LineBuilder func(v any) []string // use together with Headers and no Lines
+}
+
+type PrintRequest struct {
+	Cmd              *cobra.Command
+	Format           string
+	Fields           string
+	Annotations      map[string]string
+	Table            *Table
+	HideTableHeaders bool
```

Comment:
IMHO, **HideTableHeaders** is a property of Table itself, not PrintRequest, since the request can print other data than tables. Instread, set the Table header to **nil** and check it in the table printing logic place. The signature change of **PrintCmdOutputCustom** is then not necessary

Reply:
This command relies on the output module to construct the table being printed for output. As such, we don't have a means of altering the table headers from our command. I went with Peter's suggestion of adding PrintCmdOutputAdvanced to avoid changing PrintCmdOutputCustom

*https://github.com/cisco-open/fsoc/blob/e3338c589daffddf0f80facb966729c53d87cad0/cmd/optimize/events.go#L90-L93
* https://github.com/cisco-open/fsoc/blob/e3338c589daffddf0f80facb966729c53d87cad0/output/output.go#L180-L198
* https://github.com/cisco-open/fsoc/blob/e3338c589daffddf0f80facb966729c53d87cad0/output/output.go#L243-L253